### PR TITLE
Move tracing utils

### PR DESCRIPTION
**Pull request by erlingrj**
Moving tracing utilities from `lingua-franca` to `reactor-c`. Also install them to `/usr/local/bin`.

### CODE REVIEW

File: .github/workflows/build-trace-tools.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  run:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - name: Check out lingua-franca repository
```
Comment (lhstrh) [inline .github/workflows/build-trace-tools.yml:14]:
```suggestion
```

File: util/tracing/trace_to_chrome.o
Comment (lhstrh) [inline util/tracing/trace_to_chrome.o:1]: Looks like this file should not be checked in.

File: util/tracing/launch-fedsd.sh
Comment (lhstrh) [inline util/tracing/launch-fedsd.sh:1]: There are changes to this file in https://github.com/lf-lang/lingua-franca/pull/1695. Please coordinate with @ChadliaJerad.

File: util/tracing/trace_to_influxdb.o
Comment (lhstrh) [inline util/tracing/trace_to_influxdb.o:1]: Another binary file that does not belong.

File: util/tracing/trace_util.o
Comment (lhstrh) [inline util/tracing/trace_util.o:1]: Ifrm

File: util/tracing/visualization/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+__pycache__/
```
Comment (lhstrh) [inline util/tracing/visualization/.gitignore:1]:
```suggestion
__pycache__/

```

File: util/tracing/visualization/fedsd.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                f.write(fhlp.svg_string_draw_adv(row['x1'], row['y1'], label))
+
+        f.write('\n</svg>\n\n')
+
+        # Print footer
+        f.write('</body>\n')
+        f.write('</html>\n')
+
+    # Write to a csv file, just to double check
+    trace_df.to_csv('all.csv', index=True)
```
Comment (lhstrh) [inline util/tracing/visualization/fedsd.py:331]:
```suggestion
    trace_df.to_csv('all.csv', index=True)

```

File: util/tracing/visualization/fedsd_helper.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    Args:
+     * x: Int X coordinate of the dash
+     * y: Int Y coordinate of the dash
+     * label: String to draw 
+    Returns:
+     * String: the svg string of the triangle
+    '''
+    str_line1 = svg_string_draw_line(x-5, y, x+5, y, "ADV")
+    str_line2 = svg_string_draw_side_label(x, y, label)
+    return str_line1 + str_line2
```
Comment (lhstrh) [inline util/tracing/visualization/fedsd_helper.py:255]:
```suggestion
    return str_line1 + str_line2

```

**Review by lhstrh — CHANGES_REQUESTED** [review]
Looks OK modulo comments. Thanks for doing this!

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -22,6 +22,9 @@ jobs:
   build-rti:
     uses: lf-lang/reactor-c/.github/workflows/build-rti.yml@main
 
+  build-trace-tools:
+    uses: lf-lang/reactor-c/.github/workflows/build-trace-tools.yml@move-tracing-utils
```
Comment (edwardalee) [inline .github/workflows/ci.yml:26]: Should this be `@main`?

**Review by edwardalee — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 5 above)
Comment (erlingrj) [reply .github/workflows/ci.yml:26]: Yes. I will rename to main. But just want to run the CI pipeline before merging

**Comment by erlingrj** [comment]
@lhstrh do you know whats going on with the test failures? Seems related to a recent merge in `lingua franca`

**Comment by erlingrj** [comment]
I restored a branch in the main repo and now it works. But there are some non-determinsitic federated test failures here...

**Review by lhstrh — APPROVED** [review]

**Pull request merged**